server-common/src/main/java/org/apache/kafka/server/purgatory/DelayedOperation.java

```diff
@@ -41,7 +40,7 @@
  */
 public abstract class DelayedOperation extends TimerTask {
 
-    private final AtomicBoolean completed = new AtomicBoolean(false);
+    private volatile boolean completed = false;
```

Comment:
> onComplete() can be triggered by either forceComplete(), which forces calling onComplete() after delayMs if the operation is not yet completed, or tryComplete(), which first checks if the operation can be completed or not now, and if yes calls forceComplete().

Could we adjust the comment to sth like the following?

onComplete() is called from forceComplete(), which is triggered once by either expiration if the operation is not completed after delayMs, or tryComplete() if the operation can be completed now.

Reply:
Done.